### PR TITLE
Last major update to codebase for v1.4

### PR DESCRIPTION
**Pull request by sevenmatt7**
Arcade command has been updated so that it saves the high score attained by the user in the current instance of the application.

The undo command has been renamed to the delete command to make it clearer for the users. 

Logging has also been added to the codebase.

### CODE REVIEW

File: src/main/java/com/algosenpai/app/logic/command/HelpCommand.java [hunk 1]
```diff
@@ -23,18 +24,47 @@ public String execute() {
             return getDefaultMessage();
         } else {
             int index = userStats.getIndexByName(inputs.get(1));
+            userStats = userStats.parseString(Storage.loadData("UserData.txt"));
```
Comment (nwjbrandon) [inline src/main/java/com/algosenpai/app/logic/command/HelpCommand.java]: https://nuscs2113-ay1920s1.github.io/website/schedule/week6/topics.html
This code seems to have the arrowhead style code mentioned during lecture. Consider breaking the method down.

File: src/test/java/com/algosenpai/app/commands/HelpCommandTest.java [hunk 2]
```diff
@@ -56,6 +56,7 @@ void testHelpKeyPress() {
     }
     
 
+
```
Comment (nwjbrandon) [inline src/test/java/com/algosenpai/app/commands/HelpCommandTest.java]: Please don't add unnecessary spacing.

File: src/main/java/com/algosenpai/app/logic/command/HelpCommand.java [hunk 3]
```diff
@@ -23,18 +24,47 @@ public String execute() {
...
+                if (percentageStat < 40.0) {
+                    return "Try solving these problems on Kattis:\n"
+                            + "committeeassignment, pebblesolitaire";
+                } else if (percentageStat < 60.0) {
+                    return "Try solving these problems on Kattis:\n"
+                            + "pebblesolitaire2, equalsumeasy";
+                }
                 return "Try solving these problems on Kattis:\n"
-                        + "committeeassignment, pebblesolitaire";
+                        + "robotturtles, hidingchickens";
```
Comment (nwjbrandon) [inline src/main/java/com/algosenpai/app/logic/command/HelpCommand.java]: Would be better if these are stored as constants.

**Comment by nwjbrandon** [comment]
Good job making the changes! Please considering the following suggestions I made, and consider working on a non-master branch.

File: src/main/java/com/algosenpai/app/logic/command/HelpCommand.java (hunk 1 above)
Comment (modakshantanu) [reply src/main/java/com/algosenpai/app/logic/command/HelpCommand.java]: Agreed. @nwjbrandon raises an excellent point here and it cannot be ignored.

**Comment by modakshantanu** [comment]
So arcade highscore not stored in UserStats right?????!?!?!?!?

File: src/main/java/com/algosenpai/app/logic/command/utility/DeleteCommand.java [hunk 4]
```diff
@@ -5,13 +5,13 @@
 
 import java.util.ArrayList;
 
-public class UndoCommand extends Command {
+public class DeleteCommand extends Command {
```
Comment (modakshantanu) [inline src/main/java/com/algosenpai/app/logic/command/utility/DeleteCommand.java]: I see

File: src/main/java/com/algosenpai/app/logic/command/utility/HelpCommand.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+            if (percentageStat < 40.0) {
+                return responseString += "evenup, pairingsocks, coconut";
+            } else if (percentageStat > 40.0 && percentageStat < 60.0) {
+                return responseString += "throwns, integerlists, joinstrings";
+            } else if (percentageStat > 60.0) {
+                return responseString += "restaurant, ferryloading4, teque";
+            }
+        } else if (index == 3) {
+            if (percentageStat < 40.0) {
+                return responseString += "committeeassignment, pebblesolitaire";
```
Comment (modakshantanu) [inline src/main/java/com/algosenpai/app/logic/command/utility/HelpCommand.java]: Does not handle percentage == 40 ! Oops!

File: src/main/java/com/algosenpai/app/ui/Ui.java [hunk 6]
```diff
@@ -143,21 +146,23 @@ public void setLogic(Logic logic, UserStats stats, boolean wasDatafileCorrupted)
...
+    private void handleUserInput() throws IOException, FileParsingException {
         resetIdle();
         String input = userInput.getText();
         Command commandGenerated = logic.executeCommand(input);
         String response = commandGenerated.execute();
 
-        if (commandGenerated instanceof UndoCommand) {
+        if (commandGenerated instanceof DeleteCommand) {
             if (dialogContainer.getChildren().isEmpty()) {
                 idleMinutes = 60;
```
Comment (modakshantanu) [inline src/main/java/com/algosenpai/app/ui/Ui.java]: Oh is this changed to 60 from 180 for the demo?

File: src/main/java/com/algosenpai/app/utility/LogCenter.java [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        return fileHandler;
+    }
+
+    private static ConsoleHandler createConsoleHandler() {
+        ConsoleHandler consoleHandler = new ConsoleHandler();
+        consoleHandler.setLevel(currentLogLevel);
+        return consoleHandler;
+    }
+}
+
```
Comment (modakshantanu) [inline src/main/java/com/algosenpai/app/utility/LogCenter.java]: Good job with the logger!

File: src/main/java/com/algosenpai/app/logic/command/utility/HelpCommand.java (hunk 5 above)
Comment (modakshantanu) [reply src/main/java/com/algosenpai/app/logic/command/utility/HelpCommand.java]: Ignore this I fixed it

**Pull request merged**